ray-on-gke/cloudbuild.yaml

```diff
@@ -64,30 +67,82 @@ steps:
         ray job submit --working-dir ../example_ray_job_scripts \
         --address=http://127.0.0.1:8265 -- python ray_job.py \
         || ( echo "false" >> /workspace/ray_result.txt )
+    waitFor: ['create gke cluster']
 
-        terraform destroy -var=project_id=$PROJECT_ID \
+  - id: 'test jupyterhub'
+    name: 'gcr.io/$PROJECT_ID/terraform'
+    entrypoint: 'bash'
+    args:
+      - '-c'
+      - |
+        cd jupyter-on-gke/tests
+        python3 change_jupyter_config.py
+
+        cd ../
+        terraform apply \
+        -var=project_id=$PROJECT_ID \
         -var=namespace=$SHORT_SHA \
-        -var=service_account=$_USER_NAME-$SHORT_SHA-system-account \
-        -auto-approve -no-color
-    dir: 'ray-on-gke/user/'
-    waitFor: ['get kube config']
+        -var=add_auth=false \
+        -var=enable_iap_service=false \
+        -auto-approve -no-color \
+        || ( echo "false" > /workspace/jupyterhub_tf_result.txt )
+
+        kubectl wait --all pods -n $SHORT_SHA --for=condition=Ready --timeout=300s
+        kubectl get services -n $SHORT_SHA
+        kubectl get service proxy-public -n $SHORT_SHA --output jsonpath='{.status.loadBalancer.ingress[0].ip}' > /workspace/jupyterhub_host_url.txt
+        echo "HOST URL is " $(cat /workspace/jupyterhub_host_url.txt)
+
+        cd tests/
+        python3 test_hub.py $(cat /workspace/jupyterhub_host_url.txt) || ( echo "false" > /workspace/jupyterhub_test_result.txt )
```

Comment:
Question since I'm not super familiar with cloud build, but is tracking the test result in `/workspace/jupyterhub_test_result.txt ` needed? Shouldn't `python3 test_hub.py $(cat /workspace/jupyterhub_host_url.txt)` returning exit code 1 be enough for this stage to be marked failed?

Reply:
I saw you updated the checks to:
```
python3 test_hub.py $(cat /workspace/jupyterhub_host_url.txt) || ( exit 1 )
```

Will that prevent clean up of the cluster then?

Reply:
ah sorry, I missed the `allowExitCodes: 1` change